core/modules/server/server-sent-events.js

```diff
@@ -0,0 +1,74 @@
+/*\
+title: $:/core/modules/server/server-sent-events.js
+type: application/javascript
+module-type: library
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+/**
+ * @param {string} prefix
+ * Usually the plugin path, such as `plugins/tiddlywiki/tiddlyweb`.
+ * The route will match `/events/${prefix}` exactly.
+ * @param {(
+ *   request: import("http").IncomingMessage,
+ *   state,
+ *   emit: (event: string, data: string) => void,
+ *   end: () => void
+ * ) => void} handler
+ * A function that will be called each time a request
+ * comes in with the request and state from the
+ * route and an emit function to call.
+ */
+var ServerSentEvents = /** @class */ (function () {
+
+  function ServerSentEvents(prefix, handler) {
+    this.handler = handler;
+    this.prefix = prefix;
+  }
+  ServerSentEvents.prototype.getExports = function () {
+    return {
+      bodyFormat: "stream",
+      method: "GET",
+      path: new RegExp("^/events/" + this.prefix + "$"),
+      handler: this.handleEventRequest.bind(this)
+    };
+  };
+  /**
+   *
+   * @param {import("http").IncomingMessage} request
+   * @param {import("http").ServerResponse} response
+   * @param {*} state
+   */
+  ServerSentEvents.prototype.handleEventRequest = function (request, response, state) {
+    if (request.headers.accept && request.headers.accept.startsWith('text/event-stream')) {
+      response.writeHead(200, {
+        'Content-Type': 'text/event-stream',
```

Comment:
The core prefers double quotes for string constants, only using single quotes for strings that need to contain double quotes.